### PR TITLE
document portal tests and various fixes to make them complete

### PR DESCRIPTION
**Pull request by alexlarsson**
This fixes:
https://github.com/flatpak/xdg-desktop-portal/issues/230
But also some other stuff that I found when adding the tests to find the issues.

Additionally it adds some docs on the ownership of XdpInodes.

### CODE REVIEW

File: tests/test-doc-portal.c [hunk 1]
```diff
@@ -455,14 +455,78 @@ global_setup (void)
   g_assert (mountpoint != NULL);
 }
 
+static gboolean
+rm_rf_dir (GFile         *dir,
```
Comment (matthiasclasen) [inline tests/test-doc-portal.c]: Alternatively, we could bring in libglnx and use glnx_shutil_rm_rf_at

Comment (alexlarsson) [reply tests/test-doc-portal.c]: Lets do that when we use it for more than this?

File: document-portal/document-portal-fuse.c [hunk 2]
```diff
@@ -1753,16 +1753,8 @@ xdp_fuse_create (fuse_req_t             req,
         }
```
Comment (matthiasclasen) [inline document-portal/document-portal-fuse.c]: I think the commit message was meant to say "...are not really *under* our control..."

File: tests/test-doc-portal.c (hunk 1 above)
Comment (matthiasclasen) [reply tests/test-doc-portal.c]: ok

File: document-portal/document-portal-fuse.c (hunk 2 above)
Comment (alexlarsson) [reply document-portal/document-portal-fuse.c]: fixed

**Pull request merged**